### PR TITLE
[SL-2281] Feature: Metadata panel

### PR DESCRIPTION
**Pull request by freixedelo**
Adds a Metadata panel to the Object Listing page to view more information about the object

* [x] Feature

#### Jira

https://skylarkplatform.atlassian.net/browse/SL-2281

### CODE REVIEW

File: src/components/panel/panel.component.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const getTitle = (object: { [k: string]: string }, priority: string[]) => {
+    const [title] = priority.map((key) => object[key]).filter((key) => key);
+
+    return title;
+  };
+
+  const orderedKeys = ["title", "name", "uid"];
+
+  return (
+    <div className="z-50">
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:27]: make this first `div` the fixed part, then I think. you should be able to remove `fixed` in the other two places

There is a jump on the page of the content behind moving down when I open

File: src/components/panel/panel.component.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                      element !== "__typename" && (
+                        <>
+                          <h3 className="mt-4 mb-2 font-bold">{element}</h3>
+                          <div className="text-base-content">
+                            {data?.getObject[element]}
+                          </div>
+                        </>
+                      ),
+                  )}
+                <h3 className="mt-4 mb-2 font-bold">Title</h3>
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:77]: Remove this old data.getObject.title stuff

File: src/components/panel/panel.component.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  togglePanel: () => void;
+  uid: string;
+}
+
+export const Panel = ({ togglePanel, objectType, uid }: Props) => {
+  const { data } = useGetObject(objectType, {
+    uid: uid,
+  });
+
+  const getTitle = (object: { [k: string]: string }, priority: string[]) => {
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:18]: Better to do `object: Record<string, string>`

File: src/components/panel/panel.component.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import { Button } from "src/components/button";
+import { Expand } from "src/components/icons/expand.component";
+import { Pill } from "src/components/pill";
+import { Tabs } from "src/components/tabs/tabs.component";
+import { useGetObject } from "src/hooks/useGetObject";
+
+interface Props {
+  objectType: string;
+  togglePanel: () => void;
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:9]: imo `togglePanel` should be `closePanel`

File: src/components/panel/panel.component.tsx (hunk 1 above)
Comment (freixedelo) [reply src/components/panel/panel.component.tsx:29]: it doesn't work with that change. can't see the jump either

File: src/components/panel/panel.component.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import { Button } from "src/components/button";
+import { Expand } from "src/components/icons";
+import { Pill } from "src/components/pill";
+import { Tabs } from "src/components/tabs/tabs.component";
+import { useGetObject } from "src/hooks/useGetObject";
+import { formatObjectField } from "src/lib/utils";
+
+interface Props {
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:8]: rename to PanelProps please

File: src/components/panel/panel.component.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  closePanel: () => void;
+  uid: string;
+}
+
+export const Panel = ({ closePanel, objectType, uid }: Props) => {
+  const { data } = useGetObject(objectType, {
+    uid: uid,
+  });
+
+  const getTitle = (object: Record<string, string>, priority: string[]) => {
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:19]: Nice

Comment (james-wallis) [reply src/components/panel/panel.component.tsx:19]: but move it out of the `Panel` function please so it isn't rerendered

File: src/components/panel/panel.component.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    const [title] = priority.map((key) => object[key]).filter((key) => key);
+    return title;
+  };
+
+  const orderedKeys = ["title", "name", "uid"];
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:24]: also move the orderedKeys out

File: src/components/panel/panel.component.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+  return (
+    <>
+      <div
+        onClick={() => closePanel()}
+        className="fixed left-0 top-0 z-50 h-full w-3/5 bg-black bg-opacity-20 "
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:30]: Instead of `h-full` do `bottom-0`

File: src/components/panel/panel.component.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+      />
+      <section className="fixed top-0 right-0 z-50 h-full w-full overflow-y-scroll bg-white drop-shadow-md md:w-2/3 lg:w-7/12 xl:w-2/5 ">
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:32]: same as above on `h-full` do `bottom-0`

File: src/components/panel/panel.component.tsx [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+                </h1>
+              </div>
+            </div>
+
+            <Tabs tabs={["Metadata"]} />
+
+            <div className="p-4 pb-12 text-sm md:p-8">
+              {data?.getObject &&
+                Object.keys(data?.getObject).map(
+                  (element) =>
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:65]: optional: `element` -> `property` or `field`

File: src/components/panel/panel.component.tsx (hunk 9 above)
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:32]: From Vercel https://skylark-ui-git-feature-sl-2281metadatapanel-skylark-platform.vercel.app/#thread-id=XLpT4

I think only the content should scroll - not the header box. So move `overflow-y-scroll`

File: src/components/panel/panel.component.tsx [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+  return (
+    <>
+      <div
+        onClick={() => closePanel()}
+        className="fixed left-0 top-0 z-50 h-full w-3/5 bg-black bg-opacity-20 "
+      />
+      <section className="fixed top-0 right-0 z-50 h-full w-full overflow-y-scroll bg-white drop-shadow-md md:w-2/3 lg:w-7/12 xl:w-2/5 ">
+        {data && (
+          <>
+            <div className="p-4 pb-2 md:p-8">
```
Comment (james-wallis) [inline src/components/panel/panel.component.tsx:35]: The spacing, I think `p-8` is great on x axis but on `y` its too much. `md:p-8 md:py-4`? reduces the size of the header slightly which could be good for space

File: src/components/panel/panel.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+
+const Template: ComponentStory<React.FC<any>> = (args) => {
+  return (
+    <div className="flex flex-row space-x-2">
+      <Panel {...args} />
+    </div>
+  );
+};
+
+export const Default = Template.bind({
```
Comment (james-wallis) [inline src/components/panel/panel.stories.tsx:20]: In this story we should mock the GraphQL call to display data

**Review by james-wallis — CHANGES_REQUESTED** [review]
Some, mostly styling, changes. Great work!

File: src/components/panel/panel.component.tsx (hunk 9 above)
Comment (freixedelo) [reply src/components/panel/panel.component.tsx:32]: I thought about that too and make sense. moving it doesn't work. i need some deep investigation on this

File: src/components/panel/panel.component.tsx (hunk 11 above)
Comment (freixedelo) [reply src/components/panel/panel.component.tsx:35]: left `md:py-6`. let me know if you think 4 suits better

File: src/components/button/button.stories.tsx [hunk 13]
```diff
@@ -77,6 +77,12 @@ Outline.args = {
   variant: "outline",
 };
 
+export const Ghost = Template.bind({});
+Outline.args = {
```
Comment (james-wallis) [inline src/components/button/button.stories.tsx:81]: `Ghost.args`

File: src/components/objectListing/objectListing.test.tsx [hunk 14]
```diff
@@ -182,3 +197,84 @@ describe("row in edit mode", () => {
     );
   });
 });
+
+test("open panel click info icon", async () => {
```
Comment (james-wallis) [inline src/components/objectListing/objectListing.test.tsx:201]: `open metadata panel, check information and close`

**Review by james-wallis — CHANGES_REQUESTED** [review]

**Review by james-wallis — APPROVED** [review]

**Pull request merged**